utils/misc/misc.go

```diff
@@ -80,6 +91,96 @@ func LoadDestinations() ([]string, []string) {
 	return batchDestinations, customDestinations
 }
 
+func AddToInMemoryCount(customerID string, destinationType string, count int, tableType string) {
+	_, ok := RouterInMemoryJobCounts[tableType][customerID]
+	if !ok {
+		routerJobCountMutex.Lock()
+		RouterInMemoryJobCounts[tableType][customerID] = make(map[string]int)
+		routerJobCountMutex.Unlock()
+	}
+	RouterInMemoryJobCounts[tableType][customerID][destinationType] += count
```

Comment:
This code doesn't protects from reads happening while writing to the map. It only prevent concurrent writes, which is not enough.

You need a read lock every-time you try to read from the map. 

Similarly for the methods bellow.
```suggestion
	routerJobCountMutex.RLock()
	_, ok := RouterInMemoryJobCounts[tableType][customerID]
	if !ok {
		routerJobCountMutex.RUnLock()
		routerJobCountMutex.Lock()
		RouterInMemoryJobCounts[tableType][customerID] = make(map[string]int)
		routerJobCountMutex.Unlock()
		routerJobCountMutex.RLock()
	}
	RouterInMemoryJobCounts[tableType][customerID][destinationType] += count
	routerJobCountMutex.RUnlock()
```